diceRoll/diceRollUnitTests.js

```diff
@@ -104,44 +115,134 @@ function testList() {
       return true;
     }),
     new UnitTest('addCalculatorListeners()', function(app, test) {
-      app.addNumberKeyListeners = stub();
-      app.addDiceKeyListeners = stub();
-      app.addOperatorKeyListeners = stub();
+      app.addNumberKeyListeners = new Stub();
+      app.addDiceKeyListeners = new Stub();
+      app.addOperatorKeyListeners = new Stub();
       app.addCalculatorListeners();
       assert(app.addNumberKeyListeners.info.called.check(), 'addNumberKeyListeners() not called');
       assert(app.addDiceKeyListeners.info.called.check(), 'addDiceKeyListeners() not called');
       assert(app.addOperatorKeyListeners.info.called.check(), 'addOperatorKeyListeners() not called');
       return true;
     }),
     new UnitTest('addNumberKeyListeners()', function(app, test) {
-      app.context.attach = stub();
-      app.keypadPress = stub();
+      app.context.attach = new Stub();
+      app.keypadPress = new Stub();
       app.addNumberKeyListeners();
+      assert(app.keypadPress.bind.info.called.check(), 'keypadPress.bind() not called');
+      assert(app.keypadPress.bind.info.called.check(10), 'keypadPress.bind() not called 10 times');
+      assert(app.keypadPress.bind.info.args.all.check(0, app, 0), 'keypadPress.bind() num0 unexpected arguments');
```

Comment:
see my comment above. if you create a simple assert.compareArrays method, then you an just build one compare method that lets you compare all the args in 1 assert. Then it will stop on the first argument that fails (or you can make it keep checking them all and then fail at the end with a list of the values that weren't as expected.

```
assert = {
  compareArrays: function(expectedArray, actualArray, comment) {
     if (expectedArray.length !== actualArray.length) {
         //fail because they are not the same length, so they cant be equal
     }
     for (var i = 0; i < expectedArray.length; i++) {
         if (expectedArray[i] != actualArray[i]) {
             /// throw comment + '; failed on argument ' + i + ': ' + expectedArray[i] + ' is not equal to ' + actualArray[2];
         }
     }
  }
```